sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezDataAggregator.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.zpages;
+
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.trace.Status.CanonicalCode;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A data aggregator for the traceZ zPage.
+ *
+ * <p>The traceZ data aggregator compiles information about the running spans, span latencies, and
+ * error spans for the frontend of the zPage.
+ */
+@ThreadSafe
+public final class TracezDataAggregator {
+  private final TracezSpanProcessor spanProcessor;
+
+  /**
+   * Constructor for {@link io.opentelemetry.sdk.extensions.zpages.TracezDataAggregator}.
+   *
+   * @param spanProcessor collects span data.
+   */
+  public TracezDataAggregator(TracezSpanProcessor spanProcessor) {
+    this.spanProcessor = spanProcessor;
+  }
+
+  /**
+   * Returns a Set of running and completed span names for {@link
+   * io.opentelemetry.sdk.extensions.zpages.TracezDataAggregator}.
+   *
+   * @return a Set of {@link String}.
+   */
+  public Set<String> getSpanNames() {
+    Set<String> spanNames = new TreeSet<>();
```

Comment:
Should we have a cap on the size of this? running spans seems small, but worried about the size of completed spans.